File: src/main/java/com/aleosiss/shulkerportal/ShulkerPortal.java
package com.aleosiss.shulkerportal;

import carpet.CarpetExtension;
import carpet.CarpetServer;
import com.aleosiss.shulkerportal.carpet.ShulkerPortalCarpetSettings;
import com.aleosiss.shulkerportal.command.ShulkerPortalCommand;
import com.mojang.brigadier.CommandDispatcher;
import net.fabricmc.api.ModInitializer;
import net.minecraft.server.command.ServerCommandSource;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class ShulkerPortal implements ModInitializer, CarpetExtension {
	public static final Logger logger = LoggerFactory.getLogger(ShulkerPortal.class);

	public static void mixin() { return; }

	static {
		CarpetServer.manageExtension(new ShulkerPortal());
	}

	@Override
	public void onGameStarted()
	{
		CarpetServer.settingsManager.parseSettingsClass(ShulkerPortalCarpetSettings.class);
	}

	@Override
	public void registerCommands(CommandDispatcher<ServerCommandSource> dispatcher)
	{
		ShulkerPortalCommand.register(dispatcher);
	}

	@Override
	public void onInitialize() {}
}

File: src/main/java/io/aleosiss/shulkerportal/ShulkerPortal.kt
package io.aleosiss.shulkerportal

import carpet.CarpetExtension
import carpet.CarpetServer
import io.aleosiss.shulkerportal.carpet.ShulkerPortalCarpetSettings
import io.aleosiss.shulkerportal.command.ShulkerPortalCommand
import com.mojang.brigadier.CommandDispatcher
import net.fabricmc.api.ModInitializer
import net.minecraft.server.command.ServerCommandSource
import org.slf4j.LoggerFactory

class ShulkerPortal : ModInitializer, CarpetExtension {
    companion object {
        val logger = LoggerFactory.getLogger(ShulkerPortal::class.java)!!
        @JvmStatic
        fun mixin() {
            return
        }

        init {
            CarpetServer.manageExtension(ShulkerPortal())
        }
    }

    override fun onGameStarted() {
        CarpetServer.settingsManager.parseSettingsClass(ShulkerPortalCarpetSettings::class.java)
    }

    override fun registerCommands(dispatcher: CommandDispatcher<ServerCommandSource>) {
        ShulkerPortalCommand.register(dispatcher)
    }

    override fun onInitialize() {}
}